src/UTMSP/CMakeLists.txt

```diff
@@ -1,41 +1,70 @@
-find_package(Qt6 REQUIRED COMPONENTS Core)
+find_package(Qt6 REQUIRED COMPONENTS Core Network)
 find_package(Threads REQUIRED)
 
 qt_add_library(UTMSP STATIC)
+option(QGC_CONFIG_UTM_ADAPTER "Enable UTM Adapter" OFF)
 
-option(CONFIG_UTM_ADAPTER "Enable UTM Adapter" OFF) # TODO: Make this QGC_CONFIG_UTM_ADAPTER
-if(CONFIG_UTM_ADAPTER)
-    message(STATUS "UTMSP is Initialized")
-    target_sources(UTMSP
-        PRIVATE
-            UTMSPAircraft.cpp
-            UTMSPAircraft.h
-            UTMSPAuthorization.cpp
-            UTMSPAuthorization.h
-            UTMSPBlenderRestInterface.cpp
+    if(QGC_CONFIG_UTM_ADAPTER)
+        add_definitions(-DQGC_CONFIG_UTM_ADAPTER)
+        add_definitions(-DQT_DEBUG)
+
+        set(CMAKE_BUILD_TYPE Debug)
+
+        message(STATUS "UTMSP is Initialized")
+
+        target_sources(UTMSP
+            PRIVATE
+            UTMSPRestInterface.h
             UTMSPBlenderRestInterface.h
-            UTMSPFlightPlanManager.cpp
+            UTMSPAuthorization.h
+            UTMSPNetworkRemoteIDManager.h
+            UTMSPAircraft.h
+            UTMSPOperator.h
             UTMSPFlightPlanManager.h
-            UTMSPManager.cpp
+            UTMSPServiceController.h
+            UTMSPVehicle.h
             UTMSPManager.h
+            UTMSPFlightDetails.h
+            UTMSPRestInterface.cpp
+            UTMSPBlenderRestInterface.cpp
+            UTMSPAuthorization.cpp
             UTMSPNetworkRemoteIDManager.cpp
-            UTMSPNetworkRemoteIDManager.h
+            UTMSPAircraft.cpp
             UTMSPOperator.cpp
-            UTMSPOperator.h
-            UTMSPRestInterface.cpp
-            UTMSPRestInterface.h
+            UTMSPFlightPlanManager.cpp
             UTMSPServiceController.cpp
-            UTMSPServiceController.h
             UTMSPVehicle.cpp
-            UTMSPVehicle.h
-    )
+            UTMSPManager.cpp
+            UTMSPFlightDetails.cpp
+        )
 
     add_custom_target(UTMSPQml
         SOURCES
-            UTMSPActivationStatusBar.qml
-            UTMSPAdapterEditor.qml
-            UTMSPMapVisuals.qml
+        UTMSPAdapterEditor.qml
+        UTMSPMapVisuals.qml
+        UTMSPActivationStatusBar.qml
+        UTMSPStateStorage.qml
+    )
+
+    target_include_directories(UTMSP
+        PUBLIC
+        services
+    )
+
+    target_link_libraries(UTMSP
+
+        PRIVATE
+        libevents_parser
```

Comment:
libevent is available in qt https://doc.qt.io/qt-6/qtwebengine-3rdparty-libevent.html, lets explore if we can migrate to get rid of submodule deps

Reply:
Removed